src/hotspot/share/ci/ciReplay.cpp

```diff
@@ -638,6 +639,11 @@ class CompileReplay : public StackObj {
         tty->print_cr("# %s", _bufptr);
       }
       skip_remaining();
+    } else if (strcmp("version", cmd) == 0) {
+      _version = parse_int("version");
+      if (_version > REPLAY_VERSION) {
+        report_error("unrecognized version");
```

Comment:
Would be nice to print both versions numbers in error message.
Also I would like to be able ignore such error and process file anyway. Is `report_error` allows it?

Reply:
> Would be nice to print both versions numbers in error message.
> Also I would like to be able ignore such error and process file anyway. Is report_error allows it?

Currently report_error() saves the error string to be printed later, so to have an error message that requires formatting, I guess I would have to allocate the string using malloc or ResourceObj memory.
Right now the only ignore flag is ReplayIgnoreInitErrors.  I could introduce something like ReplayIgnoreAllErrors , or maybe turn this error into a warning.
Christian is waiting on this version number support, so maybe I could create a separate RFE for the above suggestions?

Reply:
However, thinking again about this, it should not happen that we parse a version number that's not supported. Maybe we should keep the error as it is indeed unexpected. It should also be easy to check the replay file manually in the error case to see which version number it had. Old replay files should still work as there is no "version X" line. Maybe you should also add `|| _version < 0` on L644. 

But the RFE still makes sense to improve the error reporting and to think about a new flag to ignore all errors.

Reply:
Yes, file separate RFE.